src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -304,8 +304,13 @@ private IInvocationExpression CreateBoundCallOperation(BoundCall boundCall)
             return new LazyInvocationExpression(targetMethod, instance, isVirtual, argumentsInEvaluationOrder, _semanticModel, syntax, type, constantValue, isImplicit);
         }
 
-        private ILocalReferenceExpression CreateBoundLocalOperation(BoundLocal boundLocal)
+        private IOperation CreateBoundLocalOperation(BoundLocal boundLocal)
         {
+            if (boundLocal.Syntax.Kind() == SyntaxKind.CatchDeclaration)
+            {
+                return CreateVariableDeclaration(boundLocal);
+            }
```

Comment:
Why not have `CreateBoundCatchBlockOperation` call `CreateVariableDeclaration()` directly rather than handling declarations here?

Reply:
We need to go through Create code path to ensure the bound node gets added to the bound node to operation cache.

Reply:
Actually, sounds like this is a better suggestion given our new GetOperation approach. Let me fix this.

---
In reply to: [140611924](https://github.com/dotnet/roslyn/pull/22210#discussion_r140611924) [](ancestors = 140611924)